### PR TITLE
Feature/targetconfig warn on shadowed variables

### LINKED ISSUE #97: Variables: warn when a TargetConfig variable is shadowed by an environment variable
From `Pluma.create_target_context` method, warn when a target variable is shadowed by an environment variable when calling `variables.update()`

### PR DESCRIPTION
**Pull request by mjftw**
I added an extra commit to cut down some shorten some unit tests near the one I added too.
Not related, but may as well while I'm there.

Closes #97

### CODE REVIEW

File: pluma/cli/api.py [hunk 1]
```diff
@@ -77,6 +77,12 @@ def create_target_context(target_config_path: str) -> PlumaContext:
         target_config = PlumaConfig.load_configuration('Target config', target_config_path,
                                                        PlumaConfigPreprocessor(env_vars))
         context = TargetConfig.create_context(target_config)
+
+        for variable, env_value in ((var, val) for var, val in env_vars.items()
+                                    if var in context.variables):
+            log.warning(f'"{variable}" defined in environment variables and target config.'
+                        f'{os.linesep}Using environment: {env_value}')
```
Comment (Pixep) [inline pluma/cli/api.py:84]: Nit:
* defined in _both_ ...
* {os.linesep}    Using environment:
    * Using 4 spaces indentation here, as for other similar messages

File: tests/generic/cli/test_Pluma.py [hunk 2]
```diff
@@ -37,57 +37,52 @@ def test_Pluma_create_target_context_should_parse_target_variables():
...
-    try:
-        os.environ['console'] = 'console'
-        os.environ['port'] = '/dev/random'
-        os.environ['some_env_var'] = 'echo "from env var"'
-        run_all('variable-env-sub-tests', 'variable-env-sub-target')
-    finally:
-        os.environ.clear()
-        os.environ.update(env)
+def test_Pluma_create_target_context_variables_should_reflect_env_vars(monkeypatch):
+    monkeypatch.setenv('abc', 'def')
```
Comment (Pixep) [inline tests/generic/cli/test_Pluma.py:41]: Much cleaner!

**Review by Pixep — CHANGES_REQUESTED** [review]
Minor change regarding the output

File: pluma/cli/api.py (hunk 1 above)
Comment (mjftw) [reply pluma/cli/api.py]: I think I understood what you mean and have fixed, but let me know if I've misunderstood.

Comment (Pixep) [reply pluma/cli/api.py]: I meant
> "{variable}" defined in 
to
> "{variable}" defined in **both**

and what we probably need 4 spaces before `Using environment:` if we want to match how similar messages are displayed :)

**Review by Pixep — APPROVED** [review]
Added that one comment, otherwise g2g

File: pluma/cli/api.py (hunk 1 above)
Comment (mjftw) [reply pluma/cli/api.py]: I think that's what I've done

**Pull request merged**